client/core/ownDevice.go

```diff
@@ -282,15 +301,9 @@ func (d *Device) Own(
 	}
 
 	//ownership := d.ownership
-	var supportOtm bool
-	for _, s := range ownership.SupportedOwnerTransferMethods {
-		if s == otmClient.Type() {
-			supportOtm = true
-			break
-		}
-	}
-	if !supportOtm {
-		return MakeUnavailable(fmt.Errorf("ownership transfer method '%v' is unsupported, supported are: %v", otmClient.Type(), ownership.SupportedOwnerTransferMethods))
+	otmClient := findOTMClient(otmClients, ownership.SupportedOwnerTransferMethods)
```

Comment:
This gives you the first matching client, but you can have multiple matches. Is taking the first match always correct?

Reply:
This is fine. It is not expected that clients add there multiple otm with the same type.

Reply:
Both `otmClients` and `deviceSupportedOwnerTransferMethods` are arrays though, so even if you don't have multiple clients of the same type you can have multiple matches. 

Lets say that: 
```go
otmClients = []{client for ManufacturerCertificate, client for Self}
deviceSupportedOwnerTransferMethods = []{ManufacturerCertificate, Self}
```
`findOTMClient` will always return the first match (client for `ManufacturerCertificate` in this case). I don't know what's the expected behavior of the `Own` function, so this might now be a problem, but the function is dependent on the order of the values in the `otmClients` slice, which I think we should at least document.
Also isn't possible that in this example that owning with `ManufacturerCertificate` client fails, but owning with `Self` would succeed? However, the `Self` client won't be tried and the user of the function doesn't know which client was used and failed. I guess a custom `Error` type which would have a type of the client used would be best here, so the caller could remove it from the `otmClients` slice and retry.